src/humanPose/index.js

```diff
@@ -155,11 +156,95 @@ import * as utils from './utils.js'
         return false;
     }
 
-    function tryUpdatingPoseObject(pose, humanPoseObject) {
-        // update the object position to be the average of the pose.joints
-        // update each of the tool's positions to be the position of the joint relative to the average
-        console.log('try updating pose object', pose, humanPoseObject);
+    /**
+     * @param {Array<{x, y, z, confidence}>} input joints
+     * @return {{x: number, y: number, z: number, confidence: number}} average attributes of all
+     *         input joints
+     */
+    function averageJoints(joints) {
+        let avg = { x: 0, y: 0, z: 0, confidence: 0 };
+        for (let joint of joints) {
+            avg.x += joint.x;
+            avg.y += joint.y;
+            avg.z += joint.z;
+            avg.confidence += joint.confidence;
+        }
+        avg.x /= joints.length;
+        avg.y /= joints.length;
+        avg.z /= joints.length;
+        avg.confidence /= joints.length;
+        return avg;
+    }
+
+    /**
+     * @param {Array<{x, y, z, confidence}>} all joints
+     * @param {Array<String>} selected joint names
```

Comment:
Same thing as with Number, String refers to a boxed string whereas `string` is the primitive